src/signed_data.rs

```diff
@@ -201,24 +201,26 @@ pub(crate) fn verify_signed_data(
 }
 
 pub(crate) fn verify_signature(
-    signature_alg: &SignatureAlgorithm,
+    signature_alg: &dyn SignatureAlgorithm,
     spki_value: untrusted::Input,
     msg: untrusted::Input,
     signature: untrusted::Input,
 ) -> Result<(), Error> {
     let spki = SubjectPublicKeyInfo::from_der(spki_value)?;
     if !signature_alg
-        .public_key_alg_id
+        .public_key_alg_id()
         .matches_algorithm_id_value(spki.algorithm_id_value)
     {
         return Err(Error::UnsupportedSignatureAlgorithmForPublicKey);
     }
-    signature::UnparsedPublicKey::new(
-        signature_alg.verification_alg,
-        spki.key_value.as_slice_less_safe(),
-    )
-    .verify(msg.as_slice_less_safe(), signature.as_slice_less_safe())
-    .map_err(|_| Error::InvalidSignatureForPublicKey)
+
+    signature_alg
```

Comment:
Maybe isolate the change of name and signature of `verify()` in a separate commit, so that the `SignatureAlgorithm` type exposes a `verify_signature()` method that takes the `spki` data?

Reply:
I'm saying we could have a commit that changes the existing `SignatureAlgorithm` type's implementation of `verify()` to take the `spki` as a slice and handle the `UnparsedCertificate` mechanics, while leaving the move of `SignatureAlgorithm` from type to trait to another commit.

Reply:
oh!